File: Interview Street/Codesprint 3/nCr/nCr.cpp
#include <cstdio>

const long long MOD = 142857;

long long gcd(long long a, long long b) {
  if (b == 0) return a;
  return gcd(b, a % b);
}

long long solve(long long n, long long r) {
  if (r > n - r)
    r = n - r;

  long long num = 1;
  long long den = 1;
  for (long long i = n-r+1, j = 1; i <= n; ++i, ++j) {
    num *= i;
    den *= j;
    long long g = gcd(num, den) % MOD;
    num = (num / g) % MOD;
    den = (den / g) % MOD;
  }
  return (num / den) % MOD;
}

int main() {
  int T;
  scanf("%d", &T);

  while (T--) {
    long long n, r;
    scanf("%lld%lld", &n, &r);
    printf("%lld\n", solve(n, r));
  }
  return 0;
}

File: Interview Street/Codesprint 3/nCr/nCr.py
import sys

MOD = 142857

T = int(sys.stdin.readline())


def gcd(a, b):
    return a if b == 0 else gcd(b, a % b)


def solve(n, r):
    if r > n - r:
        r = n - r

    num = 1
    den = 1
    j = 1
    for i in xrange(n - r + 1, n + 1):
        num *= i
        den *= j
        j += 1

    return (num / den) % MOD


for t in xrange(T):
    n, r = map(int, sys.stdin.readline().split())

    print solve(n, r)
